src/components/WarningAccountModal/WarningAccountModal.tsx

```diff
@@ -0,0 +1,59 @@
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useLocation } from 'react-router';
+import { useCallback, useState } from 'react';
+
+import Button from '../Button/Button';
+
+import styles from './WarningAccountModal.module.scss';
+
+import { addQueryParam, removeQueryParam } from '#src/utils/location';
+import { useAccountStore } from '#src/stores/AccountStore';
+import { resetPassword } from '#src/stores/AccountController';
+import FormFeedback from '#components/FormFeedback/FormFeedback';
+
+const WarningAccountModal = () => {
+  const { t } = useTranslation('user');
+  const email = useAccountStore((state) => state.user?.email);
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const handleCancel = useCallback(() => {
+    navigate(removeQueryParam(location, 'u'), { replace: true });
+  }, [location, navigate]);
+  const proceedToAddPasswordClickHandler = async () => {
+    try {
+      if (email) {
+        await resetPassword(email, '');
+        navigate(addQueryParam(location, 'u', 'add-password'));
+      }
+    } catch (error: unknown) {
+      setErrorMessage(t('account.add_password_error'));
+    }
+  };
+
+  return (
+    <div className={styles.formContainer}>
+      {errorMessage && (
+        <div className={styles.formFeedback}>
+          <FormFeedback variant="error">{errorMessage}</FormFeedback>
+        </div>
+      )}
+      <h2 className={styles.heading}>{t('account.warning_title')}</h2>
+      <p className={styles.paragraph}>{t('account.warning_text')}</p>
```

Comment:
I would also make the translation keys more specific:

```suggestion
      <h2 className={styles.heading}>{t('account.delete_password_warning_title')}</h2>
      <p className={styles.paragraph}>{t('account.delete_password_warning_text')}</p>
```

Or using an object:

```suggestion
      <h2 className={styles.heading}>{t('account.delete_password_warning.title')}</h2>
      <p className={styles.paragraph}>{t('account.delete_password_warning.text')}</p>
```